doc/running-simple-app.md

```diff
@@ -0,0 +1,196 @@
+for instructions.
+
+File System Types
+------------
+
+Before we can mount the application directory, we must first generate a file system image.
+The file system image generated determines which file system type is used. There are currently
+three file systems in Mystikos. 
+
+### CPIO
+
+Choose this for small applications. The entire file system will be loaded into memory when the
+application runs. When packaging your application for production, your application directory will
+be part of the executable image.
+
+### Ext2
+
+Choose this for larger applications; EXT2 uses less memory than CPIO. The file system image is a
+separate file from the executable.
+
+Mystikos supports the following features:
+- Usage of ephemeral EXT2 file systems
+- Integrity checking of EXT2 file systems ([dm-verity](https://www.kernel.org/doc/html/latest/admin-guide/device-mapper/verity.html) compatible)
+- Encryption of EXT2 file systems ([dm-crypt](https://www.kernel.org/doc/html/latest/admin-guide/device-mapper/verity.html) compatible)
+
+Modifications to EXT2 file systems are discarded when the application terminates. 
+Currently, writes are cached in trusted memory, so the number of block writes is limited by
+available memory. This limitation may be overcome later with an encrypted backing store.
+
+To support these features, Mystikos runs the following modules within the
+trusted execution environment (TEE).
+
+- EXT2 file system driver (ext2)
+- Block integrity checking device
+- Block encryption device
+- Block cache device
+
+The EXT2 image itself resides on the untrusted file system. The TEE transitions
+into the untrusted execution environment to fetch blocks. The blocks are then
+decrypted and integrity-checked within the TEE.
+
+### Host file system (insecure)
+
+Mystikos supports mounting directories from the host file system. This option is not secured by the enclave.
+
+> Mystikos applications may use a mixture of file system types and mount types.
+
+Creating a Simple Application Image
+----------------------
+
+This section explains how to create single-file images for use with
+Mystikos.
+
+### CPIO
+
+The following command creates an EXT2 image (``cpioimage``) from the
+contents of a directory (``appdir``).
+
+```
+$ myst mkcpio appdir cpioimage
+```
+### EXT2
+
+The following command creates an EXT2 image (``ext2image``) from the
+contents of a directory (``appdir``).
+
+```
+$ myst mkext2 appdir ext2image
+```
+
+> To create a signed EXT2 image to use with a signed application package, see [sign-package.md](./sign-package.md)
+
+Loading a root file system
+--------------------------------
+
+This section explains how to use an application image as a Mystikos root file system
+(or rootfs).
+
+### Using an application image during development
+During development, an application image is passed to the ``myst exec`` command as
+follows.
+
+```
+$ myst exec-sgx <fsimage> /bin/hello
```

Comment:
In other places (and in the help for "myst exec") we use the term "rootfs". Not sure that matters.

Reply:
I'll change it for consistency.